### PR TITLE
Bitbucket Server Provider Fixes

### PR DESCRIPTION
**Pull request by MarkRx**
### **User description**
Seems this code hasn't been maintained. Contains the following fixes to minimally get it up and running:

* Fix missing import
* Fix missing attribute error when PR description is empty
* Fix diff using incorrect to/from
* Fix PR id retrieval
* Fix some errors not being propagated
* Fix description publishing not using correct url

Tested with BB Datacenter ("server" rebranded) v8.9.9

___

### **PR Type**
Bug fix

___

### **Description**
This PR includes several bug fixes and improvements for the Bitbucket Server provider:
- Added missing import for `quote_plus`.
- Introduced `get_pr_id` method to retrieve PR ID.
- Corrected base and head SHA retrieval for diffs.
- Ensured errors are raised for failed inline comment publishing.
- Fixed URL generation for relevant line links.
- Improved handling of inline comments, including multi-line and single-line comments.
- Added check for PR description attribute existence.
- Updated `publish_description` to use Bitbucket client and include reviewers.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>bitbucket_server_provider.py</strong><dd><code>Multiple bug fixes and improvements for Bitbucket Server provider.</code></dd></summary>
<hr>
      
pr_agent/git_providers/bitbucket_server_provider.py

<li>Added missing import for <code>quote_plus</code>.<br> <li> Introduced <code>get_pr_id</code> method to retrieve PR ID.<br> <li> Corrected base and head SHA retrieval for diffs.<br> <li> Ensured errors are raised for failed inline comment publishing.<br> <li> Fixed URL generation for relevant line links.<br> <li> Improved handling of inline comments, including multi-line and <br>single-line comments.<br> <li> Added check for PR description attribute existence.<br> <li> Updated <code>publish_description</code> to use Bitbucket client and include <br>reviewers.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/928/files#diff-c9ca96d14ab7a2935714944f8f377c4a9bb425efde19e66595bb58d33e9f5a40">+29/-20</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: pr_agent/git_providers/bitbucket_server_provider.py [hunk 1]
```diff
@@ -242,7 +239,7 @@ def publish_inline_comment(self, comment: str, from_line: int, file: str):
         }
 
         response = requests.post(url=self._get_pr_comments_url(), json=payload, headers=self.headers)
-        return response
+        response.raise_for_status()
```
Comment (MarkRx) [inline pr_agent/git_providers/bitbucket_server_provider.py:242]: return isn't used and is inconsistent with other providers so I removed it

File: pr_agent/git_providers/bitbucket_server_provider.py [hunk 2]
```diff
@@ -334,13 +342,14 @@ def get_commit_messages(self):
...
-            "title": pr_title
-        })
-
-        response = requests.put(url=self.bitbucket_pull_request_api_url, headers=self.headers, data=payload)
-        return response
+            "title": pr_title,
+            "reviewers": self.pr.reviewers # needs to be sent otherwise gets wiped
+        }
+        
+        self.bitbucket_client.update_pull_request(self.workspace_slug, self.repo_slug, str(self.pr_num), payload)
```
Comment (MarkRx) [inline pr_agent/git_providers/bitbucket_server_provider.py:352]: return isn't used and is inconsistent with other providers so I removed it

**Comment by mrT23** [comment]
thanks @MarkRx 
Bitbucket server support is a contribution from the community, and we rely on the community to maintain it.

have you reviewed the code suggestions ?
This seems a bug, for example:

> <details><summary>Correct the attribute check to use the <code class="notranslate">self.pr</code> object instead of the <code class="notranslate">object</code> class</summary>
> <hr>
> <p dir="auto"><strong>The <code class="notranslate">get_pr_description_full</code> method should check if the <code class="notranslate">self.pr</code> object has the <code class="notranslate">description</code> <br>attribute, not the <code class="notranslate">object</code> class.</strong></p>
> <p dir="auto"><a href="https://github.com/Codium-ai/pr-agent/pull/928/files#diff-c9ca96d14ab7a2935714944f8f377c4a9bb425efde19e66595bb58d33e9f5a40R286-R289">pr_agent/git_providers/bitbucket_server_provider.py [286-289]</a></p>
> 
> ```diff
> -if hasattr(object, "description"):
> +if hasattr(self.pr, "description"):
>      return self.pr.description
>  else:
>      return None
>  ```
> 
>     
>   
> <details><summary><b>Suggestion importance[1-10]: 10</b></summary>
> <p dir="auto">Why: The suggestion correctly identifies a bug where the wrong object was checked for the attribute, which could lead to incorrect behavior. Fixing this is crucial for the method's functionality.</p>
> </details></details>

**Comment by mrT23** [comment]
## Generating PR code suggestions


Work in progress ...<br>

**Comment by mrT23** [comment]
@MarkRx 
once you will fix the code suggestion, we can merge

**Review by mrT23 — APPROVED** [review]

**Pull request merged**